torch/testing/_internal/distributed/_tensor/common_dtensor.py

```diff
@@ -301,17 +315,17 @@ def world_size(self) -> int:
 
     @property
     def backend(self) -> str:
-        backend = "nccl" if self.device_type == "cuda" else "gloo"
+        backend = "nccl" if self.device_type == "cuda" else "hccl" if self.device_type == "hpu" else "gloo"
```

Comment:
How about reusing the `PG_BACKEND` map above?

Reply:
If we still keep those `if...else` statement, then a mapping list should be maintained and changed for each new backend added even it's out-of-tree.  

Then, is it possible to use `undefined` backend which will find corresponding device backend by device type in process group init? e.g. `nccl` for cuda, `gloo` for cpu, `hccl` for hpu and `xccl` for xpu if `hccl` and `xccl` is registered out-of-tree.

In code logic,
1. Init process group with `undefined` backend.
https://github.com/pytorch/pytorch/blob/main/torch/distributed/distributed_c10d.py#L1569 
2. Register available backend in `backend_config.get_device_backend_map()` to process group
https://github.com/pytorch/pytorch/blob/main/torch/distributed/distributed_c10d.py#L1818
3. `undefined` backend will use default device backend map , which is `gloo` for cpu, `nccl` for cuda. Out-of-tree backend will also be added to this default map by https://github.com/pytorch/pytorch/blob/main/torch/distributed/distributed_c10d.py#L254

Reply:
> If we still keep those `if...else` statement, then a mapping list should be maintained and changed for each new backend added even it's out-of-tree.
> 
> Then, is it possible to use `undefined` backend which will find corresponding device backend by device type in process group init? e.g. `nccl` for cuda, `gloo` for cpu, `hccl` for hpu and `xccl` for xpu if `hccl` and `xccl` is registered out-of-tree.
> 
> In code logic,
> 
> 1. Init process group with `undefined` backend.
>    https://github.com/pytorch/pytorch/blob/main/torch/distributed/distributed_c10d.py#L1569
> 2. By backend name to query a `backend_config` which has a `device_backend_map`.
>    https://github.com/pytorch/pytorch/blob/main/torch/distributed/distributed_c10d.py#L1816
> 
> `undefined` backend name will use default device backend map , which is `gloo` for cpu, `nccl` for cuda. Out-of-tree backend will also be added to this default map by https://github.com/pytorch/pytorch/blob/main/torch/distributed/distributed_c10d.py#L254
> 
> 3. Register available backend in `backend_config.get_device_backend_map()` to process group
>    https://github.com/pytorch/pytorch/blob/main/torch/distributed/distributed_c10d.py#L1818

This can be straight one -to-one mapping for default backends : 
https://github.com/pytorch/pytorch/pull/140536

Reply:
hi @kwen2501  please review.

Reply:
hi @kwen2501 CI ran fine, please approve after review.